File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/job/JobCrafterDirectoryEntryMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait JobCrafterDirectoryEntryMessage extends MessageDeserializer {
  val messageId = 6044

  protected trait Message extends BaseMessage with Serializable {
    val messageId = JobCrafterDirectoryEntryMessage.messageId

    val playerInfo: JobCrafterDirectoryEntryPlayerInfo.TypeImpl
    val jobInfoList: Seq[JobCrafterDirectoryEntryJobInfo.TypeImpl]
    val playerLook: EntityLook.TypeImpl

    def serialize(buf: Buffer) {
      playerInfo.serialize(buf)
      buf.writeUShort(jobInfoList.length)
      for (entry <- jobInfoList) {
        entry.serialize(buf)
      }
      playerLook.serialize(buf)
    }

  }

  protected def create(playerInfo: JobCrafterDirectoryEntryPlayerInfo.TypeImpl, jobInfoList: Seq[JobCrafterDirectoryEntryJobInfo.TypeImpl], playerLook: EntityLook.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val playerInfo = JobCrafterDirectoryEntryPlayerInfo.deserialize(buf)
    val jobInfoList = {
      val builder = collection.immutable.Seq.newBuilder[JobCrafterDirectoryEntryJobInfo.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += JobCrafterDirectoryEntryJobInfo.deserialize(buf)
      }
      builder.result()
    }
    val playerLook = EntityLook.deserialize(buf)

    create(playerInfo, jobInfoList, playerLook)
  }

}

object JobCrafterDirectoryEntryMessage extends JobCrafterDirectoryEntryMessage {
  case class MessageImpl(playerInfo: JobCrafterDirectoryEntryPlayerInfo.TypeImpl, jobInfoList: Seq[JobCrafterDirectoryEntryJobInfo.TypeImpl], playerLook: EntityLook.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(playerInfo: JobCrafterDirectoryEntryPlayerInfo.TypeImpl, jobInfoList: Seq[JobCrafterDirectoryEntryJobInfo.TypeImpl], playerLook: EntityLook.TypeImpl) = MessageImpl(playerInfo, jobInfoList, playerLook)

  def apply(playerInfo: JobCrafterDirectoryEntryPlayerInfo.TypeImpl, jobInfoList: Seq[JobCrafterDirectoryEntryJobInfo.TypeImpl], playerLook: EntityLook.TypeImpl) = create(playerInfo, jobInfoList, playerLook)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/JobCrafterDirectoryEntryMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class JobCrafterDirectoryEntryMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6044;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public JobCrafterDirectoryEntryPlayerInfo playerInfo;
    public JobCrafterDirectoryEntryJobInfo[] jobInfoList;
    public EntityLook playerLook;
    
    public JobCrafterDirectoryEntryMessage() { }
    
    public JobCrafterDirectoryEntryMessage(JobCrafterDirectoryEntryPlayerInfo playerInfo, JobCrafterDirectoryEntryJobInfo[] jobInfoList, EntityLook playerLook) {
        this.playerInfo = playerInfo;
        this.jobInfoList = jobInfoList;
        this.playerLook = playerLook;
    }
    
    @Override
    public void serialize(Buffer buf) {
        playerInfo.serialize(buf);
        buf.writeUShort(jobInfoList.length);
        for (JobCrafterDirectoryEntryJobInfo entry : jobInfoList) {
            entry.serialize(buf);
        }
        playerLook.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        playerInfo = new JobCrafterDirectoryEntryPlayerInfo();
        playerInfo.deserialize(buf);
        int limit = buf.readUShort();
        jobInfoList = new JobCrafterDirectoryEntryJobInfo[limit];
        for (int i = 0; i < limit; i++) {
            jobInfoList[i] = new JobCrafterDirectoryEntryJobInfo();
            jobInfoList[i].deserialize(buf);
        }
        playerLook = new EntityLook();
        playerLook.deserialize(buf);
    }
    
}
